src/odemis/driver/pwrmccdaq.py

```diff
@@ -0,0 +1,464 @@
+            self.device.AOut(ao_ch, data)
+
+            port, bit = MCCDevice.channel_to_port(do_ch)
+            old_bit_value = self.device.DBitIn(port, bit)
+            new_bit_value = int(pwr > 0.0)
+            # update the digital output value by using a direct digital port bit
+            if old_bit_value != new_bit_value:
+                logging.debug(f"Setting do_channel {do_ch} from {old_bit_value} to {new_bit_value}")
+                self.device.DBitOut(port, bit, new_bit_value)
+
+
+class MCCDeviceSimulator:
+    """
+    A really basic and simple interface to simulate a USB-1208LS device with
+    support for DIO pin read/write commands and a single AO write command.
+    """
+    def __init__(self):
+        # initialize values
+        self.productID = 0x0007a  # USB-1208LS
+        # to keep track of the individual bits and values
+        # self.port_a_bit_config = [0, 0, 0, 0, 0, 0, 0, 0]
+        # self.port_b_bit_config = [0, 0, 0, 0, 0, 0, 0, 0]
+        self.port_a_bit_config = [False, False, False, False, False, False, False, False]
+        self.port_b_bit_config = [False, False, False, False, False, False, False, False]
+        self.AO_channels = [0, 0]  # value (uint16) in counts to output [10-bits 0-5V]
+
+        # set default configuration
+        self.DConfig(usb_1208LS.DIO_PORTA, 0x00)  # Port A output (all LOW)
+        self.DConfig(usb_1208LS.DIO_PORTB, 0xff)  # Port B input (all HIGH)
+        self.AOut(0, 0x0)
+        self.AOut(1, 0x0)
+
+    def getManufacturer(self):
+        return "MCC"
+
+    def getProduct(self):
+        return "USB-1208LS-SIM"
+
+    def getSerialNumber(self):
+        return "fake"
+
+    def DConfig(self, port_number, bit_mask):
+        """
+        This command sets the direction of the digital bits for a port.
+        :param port_number: AUXPORT = 0x10 | Port A = 0x01 | Port B = 0x04
+        :param bit_mask (int:bit value): 0 = output | 1 = input
+        """
+        self.DOut(port_number, bit_mask)
+
+    def DIn(self, port_number):
+        """
+        :param port_number: AUXPORT = 0x10 | Port A = 0x01 | Port B = 0x04
+        :return: the value seen at the port pins
+        """
+        DIO_port = self._return_port_config(port_number)
+        return sum(v << i for i, v in enumerate(DIO_port))
+
+    def DOut(self, port_number, value):
+        """
+        This command writes data to the DIO port bits that are configured as outputs.
+        :param port_number: AUXPORT = 0x10 | Port A = 0x01 | Port B = 0x04
+        :param value: value to write to the port (0-255)
+        """
+        if value < 0 or value > 255:
+            raise ValueError("Value to set is not between 0 and 255")
+
+        if port_number == usb_1208LS.DIO_PORTA:
+            self.port_a_bit_config = [bool(value & 1 << i) for i in range(8)]
+        else:
+            self.port_b_bit_config = [bool(value & 1 << i) for i in range(8)]
```

Comment:
DOut and DIn should not write/read on the `config` register. They should affect a different memory. That said, it's a simulator, so in practice, if you write somewhere, no one will ever really know. So you could just do nothing here, and it'd be fine. 
Still, if you want to be able to call DOut and followed by DIn, to simulate a change in the value read by DIn, then this current is fine... but use a separate register from `config`.